src/test/HostActivationTests/NativeHosting/Nethost.cs

```diff
@@ -88,22 +117,59 @@ public void GetHostFxrPath_WithAssemblyPath_AppLocalFxr()
             string appLocalFxrDir = Path.Combine(sharedState.BaseDirectory, "appLocalFxr");
             Directory.CreateDirectory(appLocalFxrDir);
             string assemblyPath = Path.Combine(appLocalFxrDir, "AppLocalFxr.dll");
-            string hostFxrPath = Path.Combine(appLocalFxrDir, RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("hostfxr"));
+            string hostFxrPath = Path.Combine(appLocalFxrDir, HostFxrName);
             File.WriteAllText(assemblyPath, string.Empty);
             File.WriteAllText(hostFxrPath, string.Empty);
 
             Command.Create(sharedState.NativeHostPath, $"{GetHostFxrPath} {assemblyPath}")
                 .CaptureStdErr()
                 .CaptureStdOut()
+                .EnvironmentVariable("COREHOST_TRACE", "1")
                 .Execute()
                 .Should().Pass()
                 .And.HaveStdOutContaining($"hostfxr_path: {hostFxrPath}".ToLower());
         }
 
+        private class TestRegistryKeyOverride : IDisposable
```

Comment:
Can you please expose this to all tests and refactor the one in SDK resolution to use the same class? It does exactly the same thing.

Reply:
Pulled out into `RegisteredInstallKeyOverride` class and switched other tests to it.